### ISSUE TITLE
Failing to get default MaterialGenerator in WebGL build

### ISSUE THREAD

**Issue by MattAlanWright** [atteneder/glTFast#151]
I suspect I am mis-using the library somehow, but I have reached an impasse in a project in which I'm trying to load glTF files in at runtime in a WebGL build. I have created several glTF files from OBJ files using Cesium's [obj2gltf](https://github.com/CesiumGS/obj2gltf). These OBJs **do not** have .mtl files associated with them. This led to some rendering issues that I resolved by changing the `doubleSided` field in the glTF files to "true". I can now view these glTF files in Windows' "3D Viewer" and they appear fine. I'm no familiar enough with the glTF format to know whether this is relevant so I'm including it to be safe.

I am attempting to load these objects into my Unity scene at runtime using the following code:

```C#
        var go = new GameObject(config.name);
        var gltf = go.AddComponent<GLTFast.GltfAsset>();
        gltf.url = config.url;
```

where `config.url` is a valid URL retrieved elsewhere pointing to one of these files. In the Windows build this works just fine (more than fine, in fact: load times are much improved compared to a previous approach that loaded the models as .obj files). However, in WebGL, this produces the following error in the developer console in Chrome:

```
Exception: Could not determine default MaterialGenerator
  at GLTFast.Materials.MaterialGenerator.GetDefaultMaterialGenerator () [0x00000] in <00000000000000000000000000000000>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <00000000000000000000000000000000>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <00000000000000000000000000000000>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <00000000000000000000000000000000>:0 
 
(Filename: currently not available on il2cpp Line: -1)
```

I have tried ensuring that all shaders are available by running the Windows build, saving the tracked shader assets and adding this saved asset to the pre-loaded assets, but it made no difference.

Any help would be much appreciated.

**Comment by atteneder** [comment]
Hi,

Sorry for the inconvenience. You might have the same problem as others before. Please read [this answer](https://github.com/atteneder/glTFast/issues/149#issuecomment-789662407) and see if it solves your problem.

In short, glTFast makes assumptions like "if the URP package is installed, the MaterialGenerator for Built-In is not included in builds. So if you use it anyways, you'll get this error.

I'm not happy that users stumble over this and are not able to find a solution easily. This has to be improved definitely.

hth

**Comment by MattAlanWright** [comment]
Thanks so much for the quick response! This was exactly the issue, things are working well now.

For anyone else who stumbles on this and wants to know how I resolved it: I had, at an earlier stage, installed the Lightweight Rendering Pipeline which also requires the installation of the Universal Rendering Pipeline (maybe they're technically the same thing? I'm shaky on some of these concepts so forgive any mis-steps in this explanation). I then assigned this as my scene's scriptable rendering pipeline. I later reverted back to the Built-In Rendering Pipeline and unassigned the scriptable pipeline but left the Lightweight RP and Universal RP installed. The installations must trigger some #ifdefs that then expect the pipeline to be used when it's not. I uninstalled both pipelines and it resolved the issue.
